File: AlphaBack/UniBackEnd/src/main/java/com/esprit/unibackend/entities/Offer.java
package com.esprit.unibackend.entities;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.Set;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Offer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(length = 100000)
    private String description;

    private String company;

    @Enumerated(EnumType.STRING)
    private typeOffre type;

    private int duree;
    private Date date;


    @JsonIgnore
    @OneToMany(mappedBy = "offer", cascade = CascadeType.ALL)
    private Set<Candidacy> candidacies;


}

File: AlphaFront/src/app/models/Offer.ts
import { Candidacy } from "./Candidacy";

export enum typeOffre{
    Formation, Stage, Emploi
}
export class Offer{
    id:number | undefined;

    description!:string;
    company!:string;
    type!:typeOffre;

    duree!:number;
    date!:Date;
    candidacies:Candidacy[]=[];
}